general/forall.hpp

```diff
@@ -32,8 +32,8 @@ namespace mfem
 {
 
 // Maximum size of dofs and quads in 1D.
-const int MAX_D1D = 16;
-const int MAX_Q1D = 16;
+const int MAX_D1D = 14;
+const int MAX_Q1D = 14;
```

Comment:
I'm not a big fan of this, this is a bit of a hack, I'm okay with it if it fixes your issue, but it feels a lot like burying the problem to me. Maybe that we could instantiate a few generic versions with different `MAX`, ie one version for `D1D` and `Q1D` inferior to `4`, one version for `D1D` and `Q1D` inferior to `8`, one version for `D1D` and `Q1D` inferior to `16`.

Reply:
That sounds good to me, the issue we ran into was that the fall back version of the diffusion kernel `` PADiffusionApply3D(NE,B,G,Bt,Gt,D,X,Y,D1D,Q1D);`` was maxing out register memory with ``MAX_D1D/Q1D=16``.